### ISSUE TITLE
Issue with running pre-trained 1D Burgers' equation FNO model

### ISSUE THREAD

**Issue by sakurairihito** [pdebench/PDEBench#66]
Hello,
I am trying to use the pre-trained model for the 1D Burgers' equation using Fourier Neural Operators (FNO). I downloaded the model file 1D_Burgers_Sols_Nu1.0_FNO.pt from the official repository at https://darus.uni-stuttgart.de/dataset.xhtml?persistentId=doi:10.18419/darus-2987.
However, I encountered an issue when setting the initial_step parameter to 10, as the model did not run. I then changed initial_step to 1 and obtained the following results:

```
errs= (array(0.19543895, dtype=float32), array(2.267625, dtype=float32), array(0.24097434, dtype=float32), array(0.560725, dtype=float32), array(0.19770406, dtype=float32), array([6.2055003e-02, 1.6819368e-04, 1.6790342e-06], dtype=float32))
Done.
```

These error values are an order of magnitude smaller than those reported in the paper.
I have two questions regarding this issue:

1. Why is the default initial_step set to 10 in the provided code? What is the role of the initial_step parameter in the context of this model?
2. I suspect there might be an error in the code I used to run the pre-trained model. Could you please provide an example of the correct code to execute the pre-trained model and reproduce the results from the paper?

The code I used to run the pre-trained model is attached here.

[Trained_1D_Burgers.ipynb.zip](https://github.com/pdebench/PDEBench/files/15376488/Trained_1D_Burgers.ipynb.zip)

or gist link is here, please download it: https://gist.github.com/sakurairihito/941580eee2c0558ba40b70f5762ed51f

Thank you.

**Comment by kmario23** [comment]
@sakurairihito, thanks for the report. I have tried recreating your issue on my end. And I encounter the same issue when we set `initial_step=10`. So, looking at the shape of the weight matrix of the so-called *lifting layer* (`fc0`) in the model dict of the said pretrained model, it's clear they have set the `initial_step=1`. The [channel promotion linear layer](https://github.com/pdebench/PDEBench/blob/main/pdebench/models/fno/fno.py#L89) transforms the `initial_step*num_channels+1` channels to the so-called `width`. Let's consider the `initial_step=1`. Since Burgers' equation has a single channel (field variable), `num_channels=1`. And we add 1 to this multiplication since [FNO also concatenates the sampling points (aka. grid) as an additional channel](https://github.com/pdebench/PDEBench/blob/main/pdebench/models/fno/fno.py#L105). So, we will end up with `2`, which is also the dimension 1 of the weight matrix shape of `fc0`.

Now, regarding the influence of `initial_step` on the final error metrics: Empirically, ML models tend to perform well when given a certain temporal context (here denoted as `initial_step`). So, I believe the reported numbers in the PDEBENCH paper uses `initial_step=10` and hence the lower error values achieved as well as the config files providing this setting. But they might have also tried out `initial_step=1`and that might have been released as the pretrained model under question.

After https://github.com/pdebench/PDEBench/issues/51, we have updated the data of Burgers PDE and the results in the latest arXiv version of the PDEBENCH paper. Therefore, I suggest training a model from scratch.

To summarize, `initial_step` indicates how much temporal context $u(t-N, \ldots,  t-4, t-3, t-2, t-1, t)$ we would like to feed the ML model so that it can predict future states $u(t + \Delta t)$, or sometimes purely the difference $\Delta t$, as accurately as possible.

I'll close the issue for now. Feel free to reopen tho' if needed.
